src/ts/components/subtitletogglebutton.ts

```diff
@@ -57,7 +64,25 @@ export class SubtitleToggleButton extends ToggleButton<ToggleButtonConfig> {
         });
     }
 
+    private initPlayerEvents = () => {
+        this.player.on(this.player.exports.PlayerEvent.SubtitleAdded, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.SubtitleRemoved, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.PeriodSwitch, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, this.checkSubtitleAvailability);
+    }
+
     private checkSubtitleAvailability = () => {
-        this.player.subtitles.list().length > 0 ? this.show() : this.hide();
+        const storedSubtitle: StoredSubtitleLanguage = StorageUtils.getObject(DummyComponent.instance().prefixCss('subtitlelanguage'));
+        const subtitleList = this.player.subtitles.list();
+
+        // only shows the button when subtitles are existing
+        subtitleList.length > 0 ? this.show() : this.hide();
+
+        // if the stored subtitle is set active and available, that subtitle is enabled
+        let subtitleToActivate: SubtitleTrack;
+        if (storedSubtitle && storedSubtitle.active && (subtitleToActivate = subtitleList.find(subtitle => subtitle.lang === storedSubtitle.language))) {
```

Comment:
This if statement does a bit much and assignments in if statements are always frowned upon.
If you are worried about nesting if statements you can always use early returns:
```ts
        if (!storedSubtitle || !storedSubtitle.active) {
            return;
        }

        // if the stored subtitle is set active and available, that subtitle is enabled
        const subtitleToActivate: SubtitleTrack = subtitleList.find(subtitle => subtitle.lang === storedSubtitle.language);
        if (subtitleToActivate) {
            this.on();
            this.player.subtitles.enable(subtitleToActivate.id);
        }
```